lib/ruote/mon/storage.rb

```diff
@@ -108,11 +107,12 @@ def put(doc, opts={})
       end
 
       r = begin
-        collection(doc).update(
-          { '_id' => doc['_id'], '_rev' => original['_rev'] },
+        c = collection(doc).find({ '_id' => doc['_id'], '_rev' => original['_rev'] })
+        c.update_one(
           to_mongo(opts[:update_rev] ? Ruote.fulldup(doc) : doc),
-          :safe => true, :upsert => original['_rev'].nil?)
-      rescue Mongo::OperationFailure
+          :safe => true, :upsert => original['_rev'].nil?
```

Comment:
I don't expect `:safe` is an option in newer versions. Should this be a write concern (`:w`?) instead?